packages/node_modules/@node-red/nodes/core/network/10-mqtt.js

```diff
@@ -158,9 +158,16 @@ module.exports = function(RED) {
             if(!keys || !keys.length) return null;
             keys.forEach(key => {
                 let val = srcUserProperties[key];
-                if(typeof val == "string") {
+                if(typeof val === "string") {
                     count++;
                     _clone[key] = val;
+                } else {
+                    count++;
+                    try {
+                        _clone[key] = JSON.stringify(val)
+                    } catch  (err) {
+                        console.log('MQTT v5 Property value can not be converted to a String')
+                    }
                 }
```

Comment:
```suggestion
                } else if (val !== undefined && val !== null) {
                    try {
                        _clone[key] = JSON.stringify(val)
                        count++;
                    } catch  (err) {
                        // Silently drop property
                    }
                }
```

I don't think we should pass through `null` or `undefined` if the intention here is to pass through numeric values by encoding them.

Also, moved the `count` so it only increments if the property successfully encodes.

Finally, removed the log as we shouldn't be using `console` in the node as it won't get picked up by the logger. But nor do we have a handle to the node to log against in this function. Given it already silently dropped non-string values, we can continue silently dropping things that don't encode.

Reply:
Fair